www/pfSense-pkg-vHosts/files/usr/local/pkg/vhosts.inc

```diff
@@ -0,0 +1,367 @@
+
+        location = /favicon.ico  # Do not log requests for favicon.ico.
+            {
+            return               204;
+            access_log           off;
+            log_not_found        off;
+            }
+        }
+CONF_SERVER;
+
+/*-----------------------------------------*/
+/* nginx server SSL config block template. */
+/*                                         */
+/* @crt_file_path = Certificate file path. */
+/* @key_file_path = Private key file path. */
+/*-----------------------------------------*/
+$vhosts_g['conf_server_ssl'] = <<<CONF_SERVER_SSL
+
+        ssl_certificate                         '@crt_file_path';
+        ssl_certificate_key                     '@key_file_path';
+        ssl_session_timeout                     10m;
+        keepalive_timeout                       70;
+        ssl_session_cache                       shared:SSL:10m;
+        ssl_protocols                           TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers                             "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+        ssl_prefer_server_ciphers               on;
+        add_header Strict-Transport-Security    "max-age=31536000";
+        add_header X-Content-Type-Options       nosniff;
+        ssl_session_tickets                     off;
+        ssl_dhparam                             /etc/dh-parameters.4096;
+CONF_SERVER_SSL;
+
+/*******************************************************************************
+* vhosts_build_service_config */
+/**
+* Rebuilds nginx configuration for vhosts. Called for package synchronization
+* and when any vhost is modified.
+*
+* Config file setting: <custom_php_resync_config_command>
+*******************************************************************************/
+function vhosts_build_service_config() {
+	global $config, $vhosts_g;
+
+	$conf_file_path   = $vhosts_g['conf_file_path'];
+	$conf_file_prefix = $vhosts_g['conf_file_prefix'];
+
+	/*-------------------------------------*/
+	/* Remove all existing configurations. */
+	/*-------------------------------------*/
+	conf_mount_rw();
+	exec("/bin/rm -f $conf_file_prefix*");
+
+	/*---------------------------------------*/
+	/* Build 'server' section for each host. */
+	/*---------------------------------------*/
+	$servers = '';
+	$vhosts  = $config['installedpackages']['vhosts']['config'];
+
+	for ($i=0; $i < count($vhosts); $i++) {
+		$vhost = $vhosts[$i];
+
+		/*---------------------------*/
+		/* Skip if host is disabled. */
+		/*---------------------------*/
+		if (isset($vhost['disabled'])) {
+			continue;
+		}
+
+		/*--------------------------------------------------*/
+		/* Create the root folder and a default index page. */
+		/*--------------------------------------------------*/
+		$root_folder = "{$vhosts_g['root_base_path']}/{$vhost['dirname']}";
+
+		safe_mkdir($root_folder);
+
+		if (!file_exists("$root_folder/index.php")) {
+			file_put_contents("$root_folder/index.php", "<?php\necho phpinfo();\n?>\n");
```

Comment:
This was a holdover from the original package. I like it because it lets me know PHP is working on that server. But, again, this is a case of protecting the IT director from himself. This page lets him know the server is up. He should absolutely disable the vhost until valid index page is implemented. (He/him are general pronouns, here, and not intended to be gender specific. :))